File: backend/app/schemas/reading.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from app.models.book import BookStatus


class BookCreate(BaseModel):
    title: str
    author: Optional[str] = None
    cover_url: Optional[str] = None
    status: BookStatus = BookStatus.WANT_TO_READ
    total_chapters: Optional[int] = None
    notes: Optional[str] = None


class BookUpdate(BaseModel):
    title: Optional[str] = None
    author: Optional[str] = None
    cover_url: Optional[str] = None
    status: Optional[BookStatus] = None
    progress: Optional[int] = None
    current_chapter: Optional[int] = None
    notes: Optional[str] = None


class BookResponse(BaseModel):
    id: int
    weread_id: Optional[str]
    title: str
    author: Optional[str]
    cover_url: Optional[str]
    status: BookStatus
    progress: int
    total_chapters: Optional[int]
    current_chapter: Optional[int]
    notes: Optional[str]
    last_read_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class BookListResponse(BaseModel):
    total: int
    items: list[BookResponse]


class ReadingStats(BaseModel):
    total_books: int
    want_to_read: int
    reading: int
    finished: int
    avg_progress: float

File: frontend/src/api/reading.ts
import api from './index'

export interface Book {
  id: number
  title: string
  author?: string
  status: 'want_to_read' | 'reading' | 'finished'
  progress: number
  current_chapter: number
  total_chapters: number
  created_at: string
  updated_at: string
}

export interface ReadingStats {
  total_books: number
  reading: number
  finished: number
  avg_progress: number
}

export interface BookListResponse {
  items: Book[]
  total: number
  page: number
  size: number
}

export const readingApi = {
  listBooks(params: { status?: string; page?: number; size?: number }) {
    return api.get<any, BookListResponse>('/api/reading/books', { params })
  },
  createBook(data: { title: string; author?: string; status?: string }) {
    return api.post<any, Book>('/api/reading/books', data)
  },
  updateBook(id: number, data: { status?: string; progress?: number; current_chapter?: number; total_chapters?: number }) {
    return api.put<any, Book>(`/api/reading/books/${id}`, data)
  },
  deleteBook(id: number) {
    return api.delete(`/api/reading/books/${id}`)
  },
  getStats() {
    return api.get<any, ReadingStats>('/api/reading/stats')
  },
  syncWeread() {
    return api.post('/api/reading/sync')
  }
}
